package/MDAnalysis/core/universe.py

```diff
@@ -268,13 +290,28 @@ def _generate_from_topology(self):
                 np.arange(self._topology.n_segments), self)
 
         # Update Universe namespace with segids
-        for seg in self.segments:
-            if hasattr(seg, 'segid') and seg.segid:
-                if seg.segid[0].isdigit():
-                    name = 's' + seg.segid
+        # Many segments can have same segid, so group together first
+        try:
+            # returns dict of segid:segment
+            segids = self.segments.groupby('segids')
+        except AttributeError:
+            # no segids, don't do this step
+            pass
+        else:
+            for segid, segment in segids.items():
+                if not segid:  # ignore blank segids
+                    continue
+
+                # cannot start attribute with number
+                if segid[0].isdigit():
+                    # prefix 's' if starts with number
+                    name = 's' + segid
                 else:
-                    name = seg.segid
-                self.__dict__[name] = seg
+                    name = segid
+                # if len 1 SegmentGroup, convert to Segment
+                if len(segment) == 1:
+                    segment = segment[0]
+                self.__dict__[name] = segment
```

Comment:
Yeah pretty much what you'd expect. That attribute would stop working. Unless it's a property you can't set, in which case you'd get some weird AttributeError